WordPress/gutenberg
# Font Library: Make notices more consistent

## What?

This PR attempts to improve the way notices are handled in the Font Library modal, by passing the `notice` up to the `TabPanelLayout` container component.

This PR also includes a change to make the notices manually dismissible, in line with this comment: https://github.com/WordPress/gutenberg/pull/55975#issuecomment-1820584893

Please note that local-fonts.js has been renamed upload-fonts.js, as local-fonts.js was just a container with a spacer that imported upload-fonts.js. The spacer is no longer needed, as it looks like it was only there to provide spacing between the notice and the upload component, and this is now handled in `TabPanelLayout`. See further information here: https://github.com/WordPress/gutenberg/pull/58180#discussion_r1464914306.

## Why?

This makes all the notices in the modal appear in the same place and means that they are all handled via `TabPanelLayout`, rather than separately by each component. It also makes the notices easier to update as there is less code to change.

## How?

- `Notice` component has been removed from `FontCollection`, `InstalledFonts`, and `LocalFonts`
- `Notice` component has been added to `TabPanelLayout` component
- Notice logic is still the same for `FontCollection`, `InstalledFonts`, and `LocalFonts`, but the state has been moved from the local state of each component to the global Font Library state

## Testing Instructions




1. Open the Font Library modal
2. Trigger various notices in all the tabs (Library, Upload, Font Collection)
3. Ensure all notices appear correctly and in the same place

The easiest notice to show is the success notice when a new font is installed. Alternatively, force a notice to show in the `TabPanelLayout` component, for example by adding this just before the return on line 29:

```
notice = {
	type: 'success',
	message: 'This is a test notice',
};
```

## Screenshots or screencast 

### Library Tab
| Before | After |
| ------ | ---- |
| ![image](https://github.com/WordPress/gutenberg/assets/1645628/8986de65-f99a-47b5-850f-2ed352f3c8ca) | ![image](https://github.com/WordPress/gutenberg/assets/1645628/d780ac3b-dd7f-40f7-bc2f-eba91ca7b5b5) |

### Upload Tab
| Before | After |
| ------ | ---- |
| ![image](https://github.com/WordPress/gutenberg/assets/1645628/383940fe-a49f-49c2-9108-5eae30bd680e) | ![image](https://github.com/WordPress/gutenberg/assets/1645628/d97cd7df-971d-499f-aff7-ddbe96739d3b) |

### Font Collection Tab
| Before | After |
| ------ | ---- |
| ![image](https://github.com/WordPress/gutenberg/assets/1645628/8e974d9c-774f-4261-95a5-3c57441b26b8) | ![image](https://github.com/WordPress/gutenberg/assets/1645628/1e3774bc-ee21-4939-96e1-43f07a956787) |
